### PR TITLE
Add Joined class

### PR DESCRIPTION
**Pull request by dykov**

### CODE REVIEW

File: src/main/java/io/github/dgroup/enumerable4j/Joined.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    /**
+     * Ctor.
+     * If the source predicate is null, it returns true-returned predicate instead.
+     * @param first The source predicate.
+     * @param other The source predicates.
+     */
+    @SafeVarargs
+    @SuppressWarnings("PMD.ConstructorOnlyInitializesOrCallOtherConstructors")
+    public Joined(final Predicate<X> first, final Predicate<X>... other) {
+        this.prd = val -> true;
```
Comment (dgroup) [inline src/main/java/io/github/dgroup/enumerable4j/Joined.java:51]:
```java
this.fnc = val -> {
    ... //
}

...
     public boolean test(final X val) {
           return this.fnc.apply(val);
     }
```

File: src/main/java/io/github/dgroup/enumerable4j/Joined.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * Ctor.
+     * If the source predicate is null, it returns true-returned predicate instead.
+     * @param first The source predicate.
+     * @param other The source predicates.
+     */
+    @SafeVarargs
+    @SuppressWarnings("PMD.ConstructorOnlyInitializesOrCallOtherConstructors")
+    public Joined(final Predicate<X> first, final Predicate<X>... other) {
```
Comment (dgroup) [inline src/main/java/io/github/dgroup/enumerable4j/Joined.java:50]: Please add and use in all places in Enumerable interface

Comment (dgroup) [inline src/main/java/io/github/dgroup/enumerable4j/Joined.java:50]: Please add and use in all places in Enumerable interface

File: src/main/java/io/github/dgroup/enumerable4j/Joined.java (hunk 1 above)
Comment (dgroup) [inline src/main/java/io/github/dgroup/enumerable4j/Joined.java:51]:
```java
    private final Function<X, Boolean> fnc;

     public Joined(final Predicate<X> first, final Predicate<X> ... other) {
         this.fnc = val -> {
             ... //
         }
     }

     @Override
     public boolean test(final X val) {
           return this.fnc.apply(val);
     }
```

**Review by dgroup — CHANGES_REQUESTED** [review]

**Comment by dykov** [comment]
@dgroup , please check propesed API

**Review by dgroup — APPROVED** [review]

**Comment by dgroup** [comment]
@rultor merge

**Pull request merged**